power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -260,7 +260,12 @@ struct ComponentToMathCoupling {
 
 // change of update cause topology and param change, or just param change
 struct UpdateChange {
-    bool topo, param;
+    bool topo{};
+    bool param{};
+
+    static constexpr UpdateChange operator||(UpdateChange const& a, UpdateChange const& b) {
```

Comment:
Should it be a `friend` keyword instead of `static`?

Reply:
Ah, I see your comment that `static` is also good.

Reply:
Hmmm, `clang` says `operator||` cannot `static`.

Reply:
clang does not like this

```
     /home/runner/work/power-grid-model/power-grid-model/power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp: In member function ‘void power_grid_model::MainModelImpl<power_grid_model::container_impl::ExtraRetrievableTypes<TR ...>, power_grid_model::ComponentList<ComponentType ...> >::update_component(ForwardIterator, ForwardIterator, const std::vector<power_grid_model::Idx2D>&)’:
      /home/runner/work/power-grid-model/power-grid-model/power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp:254:63: error: no match for ‘operator||’ (operand types are ‘power_grid_model::UpdateChange’ and ‘power_grid_model::UpdateChange’)
        254 |                 cached_state_changes_ = cached_state_changes_ || changed;
            |                                         ~~~~~~~~~~~~~~~~~~~~~ ^~ ~~~~~~~
            |                                         |                        |
            |                                         |                        power_grid_model::UpdateChange
            |                                         power_grid_model::UpdateChange
      /home/runner/work/power-grid-model/power-grid-model/power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp:254:63: note: candidate: ‘operator||(bool, bool)’ (built-in)
        254 |                 cached_state_changes_ = cached_state_changes_ || changed;
            |                                         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
```

will repro locally using WSL to fix

Reply:
oh right, `operatorX` **must** be friend because it's a special function 😅